core/utils.c

```diff
@@ -136,114 +136,37 @@ int utils_textToFloat(const uint8_t * buffer,
                       double * dataP,
                       bool allowExponential)
 {
-    double result;
-    int sign;
-    int i;
-
-    if (0 == length) return 0;
+    int ret = 0;
 
-    if (buffer[0] == '-')
-    {
-        sign = -1;
-        i = 1;
-    }
-    else
-    {
-        sign = 1;
-        i = 0;
+    if (length == 0) {
+        return 0;
     }
 
-    /* Must have a decimal digit first after optional sign */
-    if (i >= length || !isdigit(buffer[i])) return 0;
+    char *const buffer_c_str = lwm2m_malloc(length + 1);
```

Comment:
I realize this has already been merged, but I would prefer to see utils_textToFloat operate without the dynamic memory allocation. Dynamic memory adds a possible failure that is very real on a constrained device.

Reply:
Would have preferred it too, but could not see a realistic way to get the conversion working properly (i.e. passing the existing unit tests) on all architectures.